### PR TITLE
fix: Change infinite piece added event to infinite pieces updated

### PR DESCRIPTION
**Pull request by RasmusAlbrektsen**
This PR is a fix to the breaking change made to Sofie-Server, where the `INIFITE_PIECE_ADDED`-event was changed to `INFINITE_PIECES_UPDATED` and instead includes the array of pieces that has been updated.

### CODE REVIEW

**Review by andr9528 — APPROVED** [review]

File: src/app/core/parsers/zod-rundown-event-parser.service.spec.ts [hunk 1]
```diff
@@ -192,12 +202,12 @@ describe(ZodRundownEventParser.name, () => {
     it('does not parse a partial rundown infinite piece added event', () => {
```
Comment (LindvedKrvang) [inline src/app/core/parsers/zod-rundown-event-parser.service.spec.ts:202]: Name should be updated as well.

File: src/app/core/parsers/zod-rundown-event-parser.service.spec.ts [hunk 2]
```diff
@@ -192,12 +202,12 @@ describe(ZodRundownEventParser.name, () => {
...
       const testee = new ZodRundownEventParser(instance(mockedEntityParser))
-      const event: Partial<RundownInfinitePieceAddedEvent> = {
-        type: RundownEventType.INFINITE_PIECE_ADDED,
+      const event: Partial<RundownInfinitePiecesUpdatedEvent> = {
+        type: RundownEventType.INFINITE_PIECES_UPDATED,
         rundownId: 'some-rundown-id',
       }
 
-      const result = (): RundownInfinitePieceAddedEvent => testee.parseInfinitePieceAddedEvent(event)
+      const result = (): RundownInfinitePiecesUpdatedEvent => testee.parseInfinitePieceAddedEvent(event)
```
Comment (LindvedKrvang) [inline src/app/core/parsers/zod-rundown-event-parser.service.spec.ts:210]: `.parseInfinitePieceAddedEvent` should also be updated.

File: src/app/core/parsers/zod-rundown-event-parser.service.ts [hunk 3]
```diff
@@ -63,13 +63,14 @@ export class ZodRundownEventParser implements RundownEventParser {
   })
 
   private readonly rundownInfinitePieceAddedEventParser = zod.object({
```
Comment (LindvedKrvang) [inline src/app/core/parsers/zod-rundown-event-parser.service.ts:65]: Name should be updated as wel.

File: src/app/core/parsers/zod-rundown-event-parser.service.ts [hunk 4]
```diff
@@ -227,7 +228,7 @@ export class ZodRundownEventParser implements RundownEventParser {
     return this.rundownSetNextEventParser.parse(event)
   }
 
-  public parseInfinitePieceAddedEvent(event: unknown): RundownInfinitePieceAddedEvent {
+  public parseInfinitePieceAddedEvent(event: unknown): RundownInfinitePiecesUpdatedEvent {
```
Comment (LindvedKrvang) [inline src/app/core/parsers/zod-rundown-event-parser.service.ts:231]: Name should be updated as well.

File: src/app/core/services/rundown-event-observer.service.ts [hunk 5]
```diff
@@ -61,9 +61,9 @@ export class RundownEventObserver {
     return this.eventObserver.subscribe(RundownEventType.SET_NEXT, this.createEventValidatingConsumer(onSetNext, this.rundownEventParser.parseSetNextEvent.bind(this.rundownEventParser)))
   }
 
-  public subscribeToRundownInfinitePieceAdded(onInfinitePieceAdded: (event: RundownInfinitePieceAddedEvent) => void): EventSubscription {
+  public subscribeToRundownInfinitePieceAdded(onInfinitePieceAdded: (event: RundownInfinitePiecesUpdatedEvent) => void): EventSubscription {
```
Comment (LindvedKrvang) [inline src/app/core/services/rundown-event-observer.service.ts:64]: Parameter name should be updated as well.

File: src/app/core/services/rundown-state.service.ts [hunk 6]
```diff
@@ -83,7 +83,7 @@ export class RundownStateService implements OnDestroy {
       this.rundownEventObserver.subscribeToRundownReset(this.resetRundownFromEvent.bind(this)),
       this.rundownEventObserver.subscribeToRundownTake(this.takePartInRundownFromEvent.bind(this)),
       this.rundownEventObserver.subscribeToRundownSetNext(this.setNextPartInRundownFromEvent.bind(this)),
-      this.rundownEventObserver.subscribeToRundownInfinitePieceAdded(this.addInfinitePieceToRundownFromEvent.bind(this)),
+      this.rundownEventObserver.subscribeToRundownInfinitePieceAdded(this.updateInfinitePiecesFromEvent.bind(this)),
```
Comment (LindvedKrvang) [inline src/app/core/services/rundown-state.service.ts:86]: `.subscribeToRundownInfinitePieceAdded` should be updated too

File: src/app/core/parsers/zod-rundown-event-parser.service.spec.ts (hunk 1 above)
Comment (RasmusAlbrektsen) [reply src/app/core/parsers/zod-rundown-event-parser.service.spec.ts]: Updated!

File: src/app/core/parsers/zod-rundown-event-parser.service.spec.ts (hunk 2 above)
Comment (RasmusAlbrektsen) [reply src/app/core/parsers/zod-rundown-event-parser.service.spec.ts]: Updated!

File: src/app/core/parsers/zod-rundown-event-parser.service.ts (hunk 3 above)
Comment (RasmusAlbrektsen) [reply src/app/core/parsers/zod-rundown-event-parser.service.ts]: This has been changed!

File: src/app/core/parsers/zod-rundown-event-parser.service.ts (hunk 4 above)
Comment (RasmusAlbrektsen) [reply src/app/core/parsers/zod-rundown-event-parser.service.ts]: Updated!

File: src/app/core/services/rundown-event-observer.service.ts (hunk 5 above)
Comment (RasmusAlbrektsen) [reply src/app/core/services/rundown-event-observer.service.ts]: I can't believe i missed all these, thanks!

File: src/app/core/services/rundown-state.service.ts (hunk 6 above)
Comment (RasmusAlbrektsen) [reply src/app/core/services/rundown-state.service.ts]: Updated!

**Review by LindvedKrvang — APPROVED** [review]

**Review by vivanov90 — APPROVED** [review]
The changes look good to me. I have a few general questions about RundownEventObserver, but I'll try to ask on to the meeting next year.

**Pull request merged**